sklearn/preprocessing/_data.py

```diff
@@ -2568,7 +2569,7 @@ def __init__(
         self.n_quantiles = n_quantiles
         self.output_distribution = output_distribution
         self.ignore_implicit_zeros = ignore_implicit_zeros
-        self.subsample = subsample
+        self.subsample = subsample if subsample is not None else np.iinfo(int).max
```

Comment:
For scikit-learn estimators constructor arguments should be stored unmodified
```suggestion
        self.subsample = subsample
```
The interpretation of the value should happen in `fit`